### ISSUE TITLE
Feature request: ask to install tutorial dependencies

### ISSUE THREAD

**Issue by isteves** [rstudio/learnr#253]
Hi! @jules32, @angela-li, and I are working on improving the `use_tutorial()` function for `usethis`, which sets up the `learnr` tutorial package format . Our two main use cases are (1) package developers that want to include tutorials as documentation and (2) course instructors that want to distribute a bundle of tutorials to students.

For course instructors, it's especially important to minimize friction related to package installs. Thus, it would be great if `run_tutorial()` would prompt the user to install packages that the tutorial depends on.  

@cderv recommended 3 options to find the dependencies in the tutorial:

- [`renv::dependencies()`](https://rstudio.github.io/renv/reference/dependencies.html) - returns a list of dependencies contained in a file/folder
- [`attachment::att_from_rmds()`](https://thinkr-open.github.io/attachment/) - returns a list of dependencies contained in a file/folder
- [`rsconnect::appDependencies()`](https://rdrr.io/cran/rsconnect/man/appDependencies.html) - returns a *full tree* of dependencies contained in a file/folder

`utils::menu()` could then be used to ask users if it's okay to install all the dependencies.

**Comment by schloerke** [comment]
This is a great idea!

Do you think `packrat::appDependencies` would work for this use case?

A difficulty that we have run into for `learnr` documents is that authors are writing _broken_ code for students to fix.  While great for teaching, this is hard to parse. https://github.com/rstudio/packrat/pull/551 fixed this by parsing per chunk vs the whole file. `learnr`'s next release will already have `packrat` as a suggested dependency to help with this deployment problem, so no new dependency will be needed.

I am open to a PR.  Maybe `install_tutorial_dependencies(file)`?

Please do not include `graphics` for `utils::menu`.

-----------

Thank you for the help with `usethis::use_tutorial()`!

**Comment by isteves** [comment]
It looks like `packrat:::fileDependencies()` will work for this. I hadn't thought of broken code, but it definitely seems like a common use-case in teaching!

I'll see what I can put together 👍 

Noted. 😄  Gabor also mentioned that it can be problematic

**Comment by cderv** [comment]
This should now be closed by #267. 
Thanks for the idea @isteves!
